### PR TITLE
Add possible element types that accept propTypes

### PR DESCRIPTION
**Pull request by LAITONEN**
Specified that PropTypes validation could also be applied to stateless or functional components, which was not clearly stated.

### CODE REVIEW

**Comment by bvaughn** [comment]
Maybe you could move this note to after the example? So the paragraph flows more smoothly?

**Comment by LAITONEN** [comment]
@bvaughn You mean at the bottom of the page or below the first code excerpt?

**Comment by bvaughn** [comment]
Below the example that it's currently above, so that the note doesn't distract from the primary point the paragraph is making. :)

**Comment by LAITONEN** [comment]
@bvaughn Done!

File: content/docs/typechecking-with-proptypes.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ Greeting.propTypes = {
 };
 ```
 
+In this example, we are using a class component, but the same functionality could also be applied to the so-called stateless or functional components.
```
Comment (eps1lon) [inline content/docs/typechecking-with-proptypes.md:33]:
```suggestion
In this example, we are using a class component, but the same functionality could also be applied to function components, or components returned from [`React.memo`](https://reactjs.org/docs/react-api.html#reactmemo) or [`React.forwardRef`](https://reactjs.org/docs/react-api.html#reactforwardref).
```

We no longer use the term functional or stateless components (misleading due to hooks). Added memo and forwardRef as other examples.

File: content/docs/typechecking-with-proptypes.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ Greeting.propTypes = {
 };
 ```
 
+In this example, we are using a class component, but the same functionality could also be applied to function components, or components returned from [`React.memo`](https://reactjs.org/docs/react-api.html#reactmemo) or [`React.forwardRef`](https://reactjs.org/docs/react-api.html#reactforwardref).
```
Comment (eps1lon) [inline content/docs/typechecking-with-proptypes.md:33]:
```suggestion
In this example, we are using a class component, but the same functionality could also be applied to function components, or components created by [`React.memo`](https://reactjs.org/docs/react-api.html#reactmemo) or [`React.forwardRef`](https://reactjs.org/docs/react-api.html#reactforwardref).
```

**Comment by eps1lon** [comment]
@LAITONEN Thanks!

**Pull request merged**